userspace/libscap/scap.c

```diff
@@ -895,144 +876,70 @@ scap_t* scap_open_plugin_int(char *error, int32_t *rc, scap_source_plugin * inpu
 	handle->m_fake_kernel_proc.args[0] = 0;
 	handle->refresh_proc_table_when_saving = true;
 
+	if((*rc = handle->m_vtable->init(handle, oargs)) != SCAP_SUCCESS)
+	{
+		snprintf(error, SCAP_LASTERR_SIZE, "%s", handle->m_lasterr);
+		scap_close(handle);
+		return NULL;
+	}
+
 	return handle;
 }
 
-#ifdef HAS_ENGINE_MODERN_BPF
-/* Temp workaround until the v-table implementation is completed. */
-scap_t* scap_open_modern_bpf_int(char *error, int32_t *rc, scap_open_args *args)
+scap_t* scap_open(scap_open_args* oargs, char *error, int32_t *rc)
 {
-	/*
-	 * Allocate the handle
-	 */
-	scap_t* handle = (scap_t*)malloc(sizeof(scap_t));
-	if(!handle)
-	{
-		snprintf(error, SCAP_LASTERR_SIZE, "error allocating the scap_t structure");
-		*rc = SCAP_FAILURE;
-		return NULL;
-	}
 
-	/*
-	 * Preliminary initializations
-	 */
-	memset(handle, 0, sizeof(scap_t));
-	handle->m_mode = SCAP_MODE_MODERN_BPF;
-	handle->m_vtable = &scap_modern_bpf_vtable;
-	handle->m_engine.m_handle = handle->m_vtable->alloc_handle(handle, handle->m_lasterr);
-	if(!handle->m_engine.m_handle)
+#ifdef CYGWING_AGENT
+	if(oargs->mode == SCAP_MODE_LIVE)
 	{
-		snprintf(error, SCAP_LASTERR_SIZE, "error allocating the engine structure");
-		*rc = SCAP_FAILURE;
-		free(handle);
+		snprintf(error,	SCAP_LASTERR_SIZE, "libscap: live mode currently not supported on Windows.");
+		*rc = SCAP_NOT_SUPPORTED;
 		return NULL;
 	}
+#endif
 
-	/*
-	 * Init handle
+	/* probably at the end of the v-table work we can use just one function
+	 * with an internal switch that selects the right vtable! For the moment
+	 * let's keep different functions.
```

Comment:
@gnosek  As I said in the PR comment this refactor should help us to unify all the engines under just one `open` method, which will assign the correct `v-table` according to the engine type, in this way we can remove a lot of duplicated code. Hoping this refactor doesn't bother your parallel v-table work, if this is the case i can revert something if you need it :)